shopify/finance-dashboard-app-private-remix-ssr/api/models/shopifyOrder/actions/update.ts

```diff
@@ -0,0 +1,72 @@
+import { applyParams, save, ActionOptions, logger } from "gadget-server";
+import { preventCrossShopDataAccess } from "gadget-server/shopify";
+import { isFullPage } from "@notionhq/client";
+import { enqueueNotionJob, getNotionOrder } from "../utils";
+import { JSONValue } from "@gadget-client/finance-dashboard-app";
+import { api } from "gadget-server";
+
+export const run: ActionRun = async ({ params, record }) => {
+  applyParams(params, record);
+  await preventCrossShopDataAccess(params, record);
+  await save(record);
+};
+
+const updateOrderInNotion = async (
+  orderId: string,
+  totalPrice: string | null,
+  jsonValCOGS: JSONValue | null,
+  jsonValMargin: JSONValue | null
+) => {
+  const price = Number(totalPrice);
+  const costOfGoods = Number(jsonValCOGS);
+  const margin = Number(jsonValMargin);
+  const response = await getNotionOrder(orderId);
```

Comment:
I think we already fetched the order in `onSuccess`!
Should we just pass it along as a param instead of refetching? Or does the notion client cache the response so it doesn't cost us any time?